distribution/docker/build.gradle

```diff
@@ -21,13 +21,33 @@ configurations {
   dockerSource
   aarch64OssDockerSource
   ossDockerSource
+  transformLog4jJar
 }
 
 dependencies {
   aarch64DockerSource project(path: ":distribution:archives:linux-aarch64-tar", configuration:"default")
   dockerSource project(path: ":distribution:archives:linux-tar", configuration:"default")
   aarch64OssDockerSource project(path: ":distribution:archives:oss-linux-aarch64-tar", configuration:"default")
   ossDockerSource project(path: ":distribution:archives:oss-linux-tar", configuration:"default")
+  transformLog4jJar project(path: ":distribution:docker:transform-log4j-config", configuration: "default")
+}
+
+/**
+ * Used in the Dockerfile template to wrap commands in a retry loop. It can't be
+ * a closure because Gradle can't effectively cache those.
+ */
+class Retry {
```

Comment:
In general I'm in favour to have class definitions living in buildSrc somewhere no matter how small they are. To keep our build scripts _a bit_ cleaner and smaller. Also these classes should change less often than the script itself.